File: core/src/main/java/com/canyon/commons/cache/StaticCache.java
package com.canyon.commons.cache;

import java.util.List;

/**
 * 静态缓存接口
 * @param <TKey>
 * @param <TValue>
 */
public interface StaticCache<TKey, TValue> {
	/**
	 * 缓存大小
	 * @return
	 */
	int size();

	/**
	 * 缓存所有对象
	 * @return
	 */
	List<TValue> all();

	/**
	 * 新增缓存
	 * @param key
	 * @param value
	 */
	void add(TKey key, TValue value);

	/**
	 * 删除缓存
	 * @param key
	 */
	void remove(TKey key);

	/**
	 * 获取缓存
	 * @param key
	 * @return
	 */
	TValue get(TKey key);

	/**
	 * 清除缓存
	 */
	void clear();
}

File: commons/src/main/kotlin/com/canyon/commons/cache/StaticCache.kt
package com.canyon.commons.cache

/**
 * 静态缓存接口
 * @param <TKey>
 * @param <TValue>
</TValue></TKey> */
interface StaticCache<TKey, TValue> {
    /**
     * 缓存大小
     * @return
     */
    fun size(): Int

    /**
     * 缓存所有对象
     * @return
     */
    fun all(): List<TValue>

    /**
     * 新增缓存
     * @param key
     * @param value
     */
    fun add(key: TKey, value: TValue)

    /**
     * 删除缓存
     * @param key
     */
    fun remove(key: TKey)

    /**
     * 获取缓存
     * @param key
     * @return
     */
    operator fun get(key: TKey): TValue?

    /**
     * 清除缓存
     */
    fun clear()
}
